### PR TITLE
test: add aws sync testcase

### PR DESCRIPTION
**Pull request by frankqianms**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/apic-extension/azext_apic_extension/tests/latest/constants.py [hunk 1]
```diff
@@ -7,4 +7,8 @@
 
 TEST_REGION = "eastus"
 # to set USERASSIGNED_IDENTITY, refer to https://learn.microsoft.com/en-us/azure/api-center/import-api-management-apis?tabs=portal#option-2-import-apis-directly-from-your-api-management-instance
-USERASSIGNED_IDENTITY = os.getenv('USERASSIGNED_IDENTITY')
\ No newline at end of file
+USERASSIGNED_IDENTITY = os.getenv('USERASSIGNED_IDENTITY')
+# aws credentials KeyVault references
+ACCESS_KEY_LINK = "https://kv-canary-franktest.vault.azure.net/secrets/AccessKey"
```
Comment (frankqianms) [inline src/apic-extension/azext_apic_extension/tests/latest/constants.py:12]: should we put these key links in github secrets?

File: src/apic-extension/azext_apic_extension/tests/latest/constants.py [hunk 2]
```diff
@@ -7,4 +7,8 @@
 
 TEST_REGION = "eastus"
 # to set USERASSIGNED_IDENTITY, refer to https://learn.microsoft.com/en-us/azure/api-center/import-api-management-apis?tabs=portal#option-2-import-apis-directly-from-your-api-management-instance
-USERASSIGNED_IDENTITY = os.getenv('USERASSIGNED_IDENTITY')
\ No newline at end of file
+USERASSIGNED_IDENTITY = os.getenv('USERASSIGNED_IDENTITY')
+# aws credentials KeyVault references
+AWS_ACCESS_KEY_LINK = "https://kv-canary-franktest.vault.azure.net/secrets/AccessKey"
```
Comment (blackchoey) [inline src/apic-extension/azext_apic_extension/tests/latest/constants.py:12]: can we configure it as an environment variable?

File: src/apic-extension/azext_apic_extension/tests/latest/test_integration_commands.py [hunk 3]
```diff
@@ -64,7 +64,25 @@ def test_integration_create_apim(self):
...
+            self.kwargs.update({
+                'integration_name': self.create_random_name(prefix='cli', length=8),
+                'usi_id': USERASSIGNED_IDENTITY,
+                'access_key_link': AWS_ACCESS_KEY_LINK,
+                'secret_access_key_link': AWS_SECRET_ACCESS_KEY_LINK,
+                'aws_region': AWS_REGION
+            })
+
+            # Attach user assigned identity with access to AWS KeyVault to API Center 
+            self.cmd('az apic update --name {s} -g {rg} --identity {{type:UserAssigned,user-assigned-identities:{{{usi_id}}}}}')
```
Comment (blackchoey) [inline src/apic-extension/azext_apic_extension/tests/latest/test_integration_commands.py:79]: minor: we may update `ApicServicePreparer` to enable user assigned identity when needed, so you don't need to repeat this line every time. Maybe something like `@ApicServicePreparer(user_assigned_identity=usi_id)`

**Review by blackchoey — APPROVED** [review]

File: src/apic-extension/azext_apic_extension/tests/latest/constants.py (hunk 1 above)
Comment (blackchoey) [reply src/apic-extension/azext_apic_extension/tests/latest/constants.py]: Yes, let's do it. Thank you.

File: src/apic-extension/azext_apic_extension/tests/latest/test_integration_commands.py (hunk 3 above)
Comment (frankqianms) [reply src/apic-extension/azext_apic_extension/tests/latest/test_integration_commands.py:79]: sounds good

**Pull request merged**